EIPS/eip-7867.md

```diff
@@ -0,0 +1,565 @@
+
+##### Removed Requirements
+
+These requirements defined in EIP-5792 are removed:
+
+> The wallet:
+>
+> * MUST NOT await for any calls to be finalized to complete the batch
+> * MUST submit multiple calls as an atomic unit in a single transaction
+> * MAY revert all calls if any call fails
+> * MUST not execute any further calls after a failed call
+> * MAY reject the request if one or more calls in the batch is expected to
+>   fail, when simulated sequentially
+
+##### Added Requirements
+
+The wallet:
+
+###### Batch Atomicity
+
+* MAY break the batch over multiple transactions.
+* MUST treat a missing batch-scope `atomicity` level as equivalent to `strict`.
+* MUST provide strict guarantees (as defined above) when the batch-scope
+  `atomicity` is `strict`.
+* MUST provide _at least_ loose guarantees (as defined above) when the
+  batch-scope `atomicity` is `loose`.
+* MUST provide _at least_ the in-order call inclusion guarantee (as defined
+  above) when the batch-scope `atomicity` is `none`.
+* MAY provide loose guarantees (as defined above) when the batch-scope
+  `atomicity` is `none`.
+* MAY provide strict guarantees (as defined above) when the batch-scope
+  `atomicity` is `loose` or `none`.
+* MUST rollback the batch if one or more critical calls (as defined above) fail.
+* MUST NOT rollback the batch if zero critical calls (as defined above) fail.
+  * In other words, if the only failures are non-critical, the successful calls
+    have to appear on chain.
+* MUST NOT execute a call (or ever allow a call to be executed) more than once.
+
+###### Flow Control
+
+* MUST treat a missing call-scope `flowControl` capability as equivalent to
+  setting `onFailure` to `rollback`.
+* MUST treat a missing call-scope `onFailure` mode as equivalent to `rollback`.
+* MUST NOT execute any calls following a failed call with `onFailure` set to
+  `halt`.
+* MUST continue to execute calls as normal following a failed call with
+  `onFailure` set to `continue`.
+
+###### Errors
+
+* MUST reject (with error code `REJECTED_LEVEL`) batches containing at least one
+  critical call if the batch requests an atomicity level that the wallet can
+  provide but the user rejected (such as might happen with an
+  [EIP-7702](./eip-7702.md) set code transaction.)
+  * Note that this only applies to user rejections specifically because of
+    atomicity. It does not change the behaviour for batches rejected for other
+    reasons. This error code MUST NOT be used for other rejection reasons.
+* MUST reject (with error code `UNSUPPORTED_LEVEL`) batches containing at least
+  one critical call if the batch requests an atomicity level the wallet cannot
+  provide for any reason other than user rejection.
+  * Wallets supporting `strict` but not `loose` SHOULD NOT reject `loose`
+    batches and SHOULD instead upgrade the request to strict atomicity.
+  * Note that a batch with exactly one call _always_ satisfies the requirements
```

Comment:
For such calls any specification of flow control capabilities is effectively ignored, as it's necessarily `strict / rollback`?